auth-tokens-filter/src/main/java/com/palantir/tokens/auth/http/BearerTokenLoggingFilter.java

```diff
@@ -0,0 +1,100 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.tokens.auth.http;
+
+import com.google.common.base.Optional;
+import com.google.common.net.HttpHeaders;
+import com.palantir.tokens.auth.AuthHeader;
+import com.palantir.tokens.auth.UnverifiedJsonWebToken;
+import javax.annotation.Priority;
+import javax.ws.rs.Priorities;
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.container.ContainerRequestFilter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.slf4j.MDC;
+
+@Priority(Priorities.AUTHORIZATION)
+public class BearerTokenLoggingFilter implements ContainerRequestFilter {
+    private static final Logger log = LoggerFactory.getLogger(BearerTokenLoggingFilter.class);
+
+    public static final String USER_ID_KEY = "userId";
+    public static final String SESSION_ID_KEY = "sessionId";
+    public static final String TOKEN_ID_KEY = "tokenId";
+
+    @Override
+    public final void filter(ContainerRequestContext requestContext) {
+        clearContext(requestContext);
+
+        String rawAuthHeader = getRawAuthHeader(requestContext);
+        if (rawAuthHeader == null) {
+            log.debug("No AuthHeader present on request.");
+            return;
+        }
+
+        try {
+            UnverifiedJsonWebToken jwt = UnverifiedJsonWebToken.of(
+                    AuthHeader.valueOf(rawAuthHeader).getBearerToken());
+
+            setUnverifiedContext(requestContext, USER_ID_KEY, jwt.getUnverifiedUserId());
+
+            Optional<String> maybeUnverifiedSessionId = jwt.getUnverifiedSessionId();
+            if (maybeUnverifiedSessionId.isPresent()) {
+                setUnverifiedContext(requestContext, SESSION_ID_KEY, maybeUnverifiedSessionId.get());
+            }
+
+            Optional<String> maybeUnverifiedTokenId = jwt.getUnverifiedTokenId();
+            if (maybeUnverifiedTokenId.isPresent()) {
+                setUnverifiedContext(requestContext, TOKEN_ID_KEY, maybeUnverifiedTokenId.get());
+            }
+        } catch (Throwable t) {
+            log.debug("Unable to process auth header.", t);
+        }
+    }
+
+    private String getRawAuthHeader(ContainerRequestContext requestContext) {
+        String rawAuthHeader = requestContext.getHeaderString(HttpHeaders.AUTHORIZATION);
+        return rawAuthHeader != null
+                ? rawAuthHeader
+                : requestContext.getHeaderString(HttpHeaders.AUTHORIZATION.toLowerCase());
+    }
+
+    private void clearContext(ContainerRequestContext requestContext) {
+        clearContext(requestContext, USER_ID_KEY);
+        clearContext(requestContext, SESSION_ID_KEY);
+        clearContext(requestContext, TOKEN_ID_KEY);
+    }
+
+    private void clearContext(ContainerRequestContext requestContext, String key) {
+        MDC.remove(key);
+        requestContext.removeProperty(getRequestPropertyKey(key));
```

Comment:
ContainerRequestContext is per-request and not ThreadLocal state, I think this remove operation is unnecessary.